feature(page-sections) Add links to individual page sections

paulmolluzzo opened the pull request:
<!--
Thanks for your interest in the project. I appreciate bugs filed and PRs submitted!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

If you're new to contributing to open source projects, you might find this free
video course helpful: http://kcd.im/pull-request

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->
**What**: Adds links to the headings of page sections.

<!-- Why are these changes necessary? -->
**Why**: Makes it easier to share specific parts of the documentation.

<!-- How were these changes implemented? -->
**How**: Added a basic function to page-sections that converts the section title to a lowercase string with dashes.


<!-- feel free to add additional comments -->

paulmolluzzo commented:
@tdeschryver Would you like to take a look at this? Thanks!

Anyone else who wants to review and give a 👍  or 👎  that'd be helpful, then I can merge it in.

tdeschryver commented:
Looks good for me! 
Later on we can make it a bit fancy with an icon when you hover over it (like the github readme)?

paulmolluzzo commented:
@tdeschryver good call on the `:hover` state. I'll get that in soon.

kentcdodds commented on components/page-sections.js:
```diff
@@ -12,6 +12,9 @@ import mdToHTML from './utils/md-to-html'
   'https://github.com/kentcdodds/glamorous-website/edit/master'
 
+const sectionTitleAsID = title =>
+  `${title.toLowerCase().replace(/ /g, '-').replace(/`/g, '')}`
```
I wonder whether it'd be safer in the long-run to use [`slugify`](https://www.npmjs.com/package/slugify) instead. Wouldn't add a TON to our bundle and would probably be nicer.

paulmolluzzo replied:
OK, I can do that for sure. Should be able to do that tonight.

paulmolluzzo commented:
A few updates:
* Merged in `master`
* Added in `slugify` to create slugs instead of the fn I had before
* Added in a link svg and used that to make a nice `:hover` effect. (see below)

![screen shot 2017-06-22 at 10 26 10 pm](https://user-images.githubusercontent.com/737065/27464280-78279a68-579a-11e7-8a00-462aadea6e28.png)

kentcdodds commented:
Love it! Thanks! I think the link svg looks a little big. How about the same size ratio that we see on GitHub README links?

paulmolluzzo commented:
How's this change I pushed?

![screen shot 2017-06-23 at 6 07 25 am](https://user-images.githubusercontent.com/737065/27477494-6bfc996a-57da-11e7-9710-bfa18e2fca73.png)
